contracts/liquidity_pool.cairo

```diff
@@ -268,58 +350,129 @@ end
+            currency_address = currency_address,
+            option_token_address = option_token_address,
+            amount = amount,
+            premia = premia,
+            fees = fees
+        )
+    else:
+        _burn_option_token_short(
+            currency_address = currency_address,
+            option_token_address = option_token_address,
+            amount = amount,
+            premia = premia,
+            fees = fees,
+            option_type = option_type,
+            underlying_price = underlying_price
+        )
+    end
 
-    # if option_side == TRADE_SIDE_LONG:
-    #     _burn_option_token_long(address, amount, strike, maturity)
-    # else:
-    #     _burn_option_token_short(address, amount, strike, maturity)
-    # end
     return ()
 end
 
 func _burn_option_token_long{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
-    address: felt,
+    currency_address: felt,
+    option_token_address: felt,
     amount: felt,
-    strike: felt,
-    maturity: felt
+    premia: felt,
+    fees: felt,
 ):
-    # address is user address
-    #long_call_burn(address, amount, strike, maturity)
+    alloc_locals
 
-    # user gets premia minus fees
-    # pool updates its available capital (unlocks it)
+    let current_contract_address = get_contract_address()
+    let user_address = get_caller_address()
+
+    # Burn the tokens
+    IOptionToken.burn(option_token_address, user_address, amount)
+
+    # Send (premia - fees)  to user
+    let to_be_received_by_user = premia - fees
+
+    IERC20.transferFrom(
+        contract_address = currency_address,
+        sender = current_contract_address,
+        recipient = user_address,
+        amount = to_be_received_by_user
+    )
 
+    # FIXME: Should happen here or in IOptionToken.burn() function?
+    # pool updates its available capital (unlocks it)
     # increase available capital by (amount - premia + fees)... (this might be happening in the amm.cairo)
     return ()
 end
 
 func _burn_option_token_short{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
-    address: felt,
+    currency_address: felt,
+    option_token_address: felt,
     amount: felt,
-    strike: felt,
-    maturity: felt
+    premia: felt,
+    fees: felt,
+    option_type: felt,
+    underlying_price: felt,
 ):
-    # address is user address
-    #short_call_burn(address, amount, strike, maturity)
+    alloc_locals
 
-    # user receives back its locked capital, pays premia and fees
-    # pool updates it available capital
+    let current_contract_address = get_contract_address()
+    let user_address = get_caller_address()
+
+    # Burn the tokens
+    IOptionToken.burn(option_token_address, user_address, amount)
+
+    # User pays (premia + fees)
+    let to_be_paid_by_user = premia + fees
 
+    # Retrieve locked capital to be paid to user
+    if option_type == OPTION_PUT:
+        let unlocked_capital_for_user = Math64x61.mul(underlying_price, amount)
```

Comment:
`underlying_price` is incorrect... locked capital is always `strike * option_size`... so there should be `strike_price` instead of the underlying price